library/core/src/iter/traits/iterator.rs

```diff
@@ -1910,11 +1932,25 @@ pub trait Iterator {
     fn try_collect<B>(&mut self) -> ChangeOutputType<Self::Item, B>
     where
         Self: Sized,
-        <Self as Iterator>::Item: Try,
-        <<Self as Iterator>::Item as Try>::Residual: Residual<B>,
-        B: FromIterator<<Self::Item as Try>::Output>,
+        <Self as Iterator>::Item: ~const Try,
+        <<Self as Iterator>::Item as Try>::Residual: ~const Residual<B> + ~const Destruct,
+        B: ~const FromIterator<<Self::Item as Try>::Output> + ~const Destruct,
     {
-        try_process(ByRefSized(self), |i| i.collect())
+        const fn collect<
+            'a,
+            T: ~const Iterator,
+            R,
+            O: ~const FromIterator<<<T as Iterator>::Item as Try>::Output>,
+        >(
+            x: GenericShunt<'a, T, R>,
+        ) -> O
+        where
+            T::Item: ~const Try<Residual = R>,
+            R: ~const Destruct,
+        {
+            x.collect()
+        }
```

Comment:
this looks terrible, can the generic constraints be moved to where bounds?

Reply:
Done